spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/CreateChangelogViewProcedure.java

```diff
@@ -179,13 +187,24 @@ private boolean shouldRemoveCarryoverRows(ProcedureInput input) {
     return input.asBoolean(REMOVE_CARRYOVERS_PARAM, true);
   }
 
-  private Dataset<Row> removeCarryoverRows(Dataset<Row> df) {
+  private Dataset<Row> removeCarryoverRows(Dataset<Row> df, boolean netChanges) {
+    Predicate<String> columnsToRemove;
+    if (netChanges) {
+      columnsToRemove =
+          column ->
+              column.equals(MetadataColumns.CHANGE_TYPE.name())
+                  || column.equals(MetadataColumns.CHANGE_ORDINAL.name())
+                  || column.equals(MetadataColumns.COMMIT_SNAPSHOT_ID.name());
+    } else {
+      columnsToRemove = column -> column.equals(MetadataColumns.CHANGE_TYPE.name());
+    }
+
     Column[] repartitionSpec =
         Arrays.stream(df.columns())
-            .filter(c -> !c.equals(MetadataColumns.CHANGE_TYPE.name()))
+            .filter(columnsToRemove.negate())
```

Comment:
if we are only using this as a negation, can we just compute the negation instead?

Reply:
I argued with myself since we each use the negative world in the variable name like `columnsNotToRemove` or use `columnsToRemove.negate()`. I don't have any preference to be honest. WDYT?

Reply:
Yea if we use set above, we can use notContains?